build.gradle

```diff
@@ -1,78 +1,265 @@
+    implementation 'org.hibernate:hibernate-core:5.6.15.Final'
+    implementation group: 'xerces', name: 'xercesImpl', version: '2.12.2'
+    implementation(group: 'org.springframework.boot', name: 'spring-boot-starter-jdbc') {
+        exclude group: 'org.apache.tomcat', module: 'tomcat-jdbc'
+    }
+
+    testImplementation libraries.junit5
+    testImplementation group: 'info.solidsoft.gradle.pitest', name: 'gradle-pitest-plugin', version: '1.4.0'
+    testImplementation group: 'org.easymock', name: 'easymock', version: '5.0.1'
+    testImplementation group: 'org.jdom', name: 'jdom2', version: '2.0.6.1'
+    testImplementation 'org.mockito:mockito-core:5.2.0'
+    testImplementation group: 'org.mockito', name: 'mockito-inline', version: '3.11.1'
+    testImplementation group: 'org.powermock', name: 'powermock-api-mockito2', version: '2.0.9'
+    testImplementation group: 'org.powermock', name: 'powermock-module-junit4', version: '2.0.9'
+    testImplementation group: 'org.dbunit', name: 'dbunit', version: '2.7.3'
+    testImplementation 'org.springframework:spring-test:6.0.7'
+    testImplementation group: 'xerces', name: 'xercesImpl', version: '2.12.2'
+    testImplementation 'org.apache.logging.log4j:log4j-slf4j2-impl:2.20.0'
+    testImplementation("org.springframework.boot:spring-boot-starter-test") {
+        exclude group: 'junit', module: 'junit'
+        exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
+    }
 
     // This dependency is exported to consumers, that is to say found on their compile classpath.
     api 'org.apache.commons:commons-math3:3.6.1'
+    api 'org.apache.commons:commons-lang3:3.12.0'
+    api 'commons-io:commons-io:2.11.0'
+}
 
-    // This dependency is used internally, and not exposed to consumers on their own compile classpath.
-    implementation 'com.google.guava:guava:29.0-jre'
+subprojects {
+    group 'com.github.hmcts'
+    version buildNumber
+    apply plugin: 'java-library'
+    apply plugin: 'idea'
+    apply plugin: "org.sonarqube"
+    apply plugin: 'org.springframework.boot'
+    apply plugin: 'io.spring.dependency-management'
+    apply plugin: 'com.github.ben-manes.versions'
+
+    configurations {
+        testUnitImplementation.extendsFrom testImplementation
+        testUnitImplementation.extendsFrom runtimeOnly
+
+        testIntegrationImplementation.extendsFrom testImplementation
+        testIntegrationImplementation.extendsFrom runtimeOnly
+    }
+
+    dependencies {
+
+        // This dependency is used internally, and not exposed to consumers on their own compile classpath.
+        implementation 'com.google.guava:guava:30.0-jre'
+        implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: log4jVersion
+        implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: log4jVersion
+        implementation 'org.apache.logging.log4j:log4j-slf4j2-impl:2.20.0'
```

Comment:
Should this use the log4jVersion property for the version number instead of the hard coded value?

Reply:
Confirmed